src/MagentoHackathon/Composer/Magento/Parser/PackageXmlParser.php

```diff
@@ -64,6 +64,8 @@ protected function parseMappings()
                     foreach ($target->children() as $child) {
                         foreach ($this->getElementPaths($child) as $elementPath) {
                             $relativePath = $basePath . '/' . $elementPath;
+                            //remove the any trailing './' or '.' from the targets base-path.
+                            $relativePath = preg_replace('#^[\/\.]*#', '', $relativePath);
```

Comment:
It didn't really need to be a `preg_match` anyway. `ltrim` will suffice, I've still moved it to a private method